hathor/utils/weight.py

```diff
@@ -0,0 +1,37 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import math
+
+
+def weight_to_work(weight: float) -> int:
+    """Convert weight to work rounding up to the nearest integer."""
+    return math.ceil(2**weight)
+
+
+def work_to_weight(work: int) -> float:
+    """Convert work to weight."""
+    if work <= 1:
+        return 0.0
+    return math.log2(work)
+
+
+def calculate_min_significant_weight(score: int, tol: float) -> float:
+    """ This function will return the min significant weight to increase score by tol.
+
+    When most peers are updated to store work as integers for their internal score and accumulated weight metadata,
+    this function will not be needed anymore. It's only use currently is to make sure miner nodes will produce blocks
+    with weight that are high enough for outdated nodes to be able to observe the score increasing.
```

Comment:
Should we use Feature Activation to remove this?

Reply:
I don't think it will be too useful. Because the peers that would "suffer any consequences" would be the peers that aren't mining. And removing the "minimum significant weight" is not technically a breaking change, because it is not a rule that must be followed, it's just the default generated weight that miners will typically use.

Also, we don't have any rush to remove it, so in the future when we observe that most peers are running `>v0.63.0` we could prepare to remove the code for the minimum significant weight.